### PR TITLE
feat(aapd-1195): check documents access via auth tokens

### PR DESCRIPTION
**Pull request by bryn-pins**
## Ticket Number

https://pins-ds.atlassian.net/browse/AAPD-1195

## Description of change

Use tokens to check document access

## Checklist

- [ ] Requires infrastructure changes
- [ ] I have updated the documentation accordingly
- [x] My commit history in this PR is linear
- [x] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `main` (please only [rebase](https://github.com/Planning-Inspectorate/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: packages/auth-server/src/account/account.js [hunk 1]
```diff
@@ -46,11 +46,19 @@ class Account {
...
 	 */ // eslint-disable-next-line no-unused-vars
 	async claims(use, scope, claims, rejected) {
-		return {
+		const details = {
 			sub: this.accountId, // it is essential to always return a sub claim
 			email: this.user.email
 		};
 
+		if (this.user.isLpaUser) {
+			details.lpaCode = this.user.lpaCode;
```
Comment (bryn-pins) [inline packages/auth-server/src/account/account.js:55]: add lpa details to id token

File: packages/document-service-api/test/developer/developer.test.js [hunk 2]
```diff
@@ -1,15 +1,74 @@
...
 	isFeatureActive: () => true
 }));
 
+jest.mock('express-oauth2-jwt-bearer', () => {
+	return {
+		auth: jest.fn(() => {
+			return (req, res, next) => {
+				req.auth = {
+					payload: {
+						sub: '29670d0f-c4b4-4047-8ee0-d62b93e91a11'
```
Comment (bryn-pins) [inline packages/document-service-api/test/developer/developer.test.js:28]: couldn't figure out how to vary this by test

File: packages/auth-server/src/account/account.js [hunk 3]
```diff
@@ -46,11 +46,19 @@ class Account {
...
+			details.isLpaAdmin = this.user.isLpaAdmin;
+			details.lpaStatus = this.user.lpaStatus;
```
Comment (Si3rr4wow) [inline packages/auth-server/src/account/account.js:57]: what is lpaStatus?

Comment (bryn-pins) [reply packages/auth-server/src/account/account.js:57]: added|confirmed|removed
can't login with removed
might not be needed for anything but thought I'd just include the lpa specific user details

File: packages/document-service-api/test/developer/developer.test.js (hunk 2 above)
Comment (Si3rr4wow) [reply packages/document-service-api/test/developer/developer.test.js:28]: I think the options are to either remock the entire function or you could try setting the auth header on the supertest request, but that might not work

Comment (Si3rr4wow) [reply packages/document-service-api/test/developer/developer.test.js:28]: Oh or we could have a common func got getting the sub from a request then we mock that function, like
```
exports.getSub = (req) => req.auth.payload.sub
```
```
jest.mock('../getSub', () => ({
  getSub: () => <some guid>
}))
```

then you can just mock the return value

```
getSub.mockReturnValueOnce(<some other guid>)
```

**Review by Si3rr4wow — APPROVED** [review]
I'd like to use the middleware you've made here so shall we get this in, I'll rebase 1502 and improve your test if I can find a nicer way to mock?

**Pull request merged**